content/en/docs/concepts/storage/volume-snapshots.md

```diff
@@ -77,58 +77,62 @@ Deletion removes both the `VolumeSnapshotContent` object from the Kubernetes API
 
 ## Volume Snapshot Contents
 
-Each VolumeSnapshotContent contains a spec, which is the specification of the volume snapshot.
+Each VolumeSnapshotContent contains a spec and status, which is the specification and status of the volume snapshot content. In dynamic provisioning, the `VolumeSnapshotContent` object is created by the snapshot common controller.
 
 ```yaml
-apiVersion: snapshot.storage.k8s.io/v1alpha1
+apiVersion: snapshot.storage.k8s.io/v1beta1
 kind: VolumeSnapshotContent
 metadata:
-  name: new-snapshot-content-test
+  name: snapcontent-72d9a349-aacd-42d2-a240-d775650d2455
 spec:
-  snapshotClassName: csi-hostpath-snapclass
+  deletionPolicy: Delete
+  driver: hostpath.csi.k8s.io
   source:
-    name: pvc-test
-    kind: PersistentVolumeClaim
-  volumeSnapshotSource:
-    csiVolumeSnapshotSource:
-      creationTime:    1535478900692119403
-      driver:          csi-hostpath
-      restoreSize:     10Gi
-      snapshotHandle:  7bdd0de3-aaeb-11e8-9aae-0242ac110002
+    volumeHandle: ee0cfb94-f8d4-11e9-b2d8-0242ac110002
+  volumeSnapshotClassName: csi-hostpath-snapclass
+  volumeSnapshotRef:
+    name: new-snapshot-test
+    namespace: default
+    uid: 72d9a349-aacd-42d2-a240-d775650d2455
 ```
 
-### Class
+For pre-provisioned snapshots, you (as cluster administrator) are responsible for creating the `VolumeSnapshotContent` object.
 
-A VolumeSnapshotContent can have a class, which is specified by setting the
-`snapshotClassName` attribute to the name of a
-[VolumeSnapshotClass](/docs/concepts/storage/volume-snapshot-classes/).
-A VolumeSnapshotContent of a particular class can only be bound to VolumeSnapshots requesting
-that class. A VolumeSnapshotContent with no `snapshotClassName` has no class and can only be bound
-to VolumeSnapshots that request no particular class.
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshotContent
+metadata:
+  name: new-snapshot-content-test
+spec:
+  deletionPolicy: Delete
+  driver: hostpath.csi.k8s.io
+  source:
+    snapshotHandle: 7bdd0de3-aaeb-11e8-9aae-0242ac110002
+  volumeSnapshotRef:
+    name: new-snapshot-test
+    namespace: default
+```
 
 ## VolumeSnapshots
 
 Each VolumeSnapshot contains a spec and a status, which is the specification and status of the volume snapshot.
 
 ```yaml
-apiVersion: snapshot.storage.k8s.io/v1alpha1
+apiVersion: snapshot.storage.k8s.io/v1beta1
 kind: VolumeSnapshot
 metadata:
   name: new-snapshot-test
 spec:
-  snapshotClassName: csi-hostpath-snapclass
+  volumeSnapshotClassName: csi-hostpath-snapclass
   source:
-    name: pvc-test
-    kind: PersistentVolumeClaim
+    persistentVolumeClaimName: pvc-test
```

Comment:
Maybe it's worth splitting up this section into "How to request a dynamically provisioned snapshot" and "How to request a pre-existing snapshot", and give examples for both.

Reply:
I'd like to submit another PR as a task page for that.  Things described at Concepts page seem to be relatively high level.

Reply:
:+1: for a separate task page.

Reply:
I guess it's a little strange that not all the fields in the VolumeSnapshot are not explained, considering that this is the user-facing object. It feels inconsistent with the VolumeSnapshotContent object above that is very detailed but actually not really that important to the user.